### ISSUE TITLE
E721: do not enforce for union or parameterised generics

### ISSUE THREAD

**Issue by jakewilliami** [astral-sh/ruff#19709]
### Summary

[E721](https://docs.astral.sh/ruff/rules/type-comparison/) is good for checking equality of simple types:
```python
>>> int is int
True
```

But struggles with order of union types (even though these are the same):
```python
>>> str | int == int | str
True
>>> str | int is int | str
False
>>> int | float | str == ((int | float) | str)
True
>>> int | float | str is ((int | float) | str)
False
```

And with parameterised generics:
```python
>>> list[int] == list[int]
True
>>> list[int] is list[int]
False
```

I wonder if Ruff should know that it can't enforce E721 for all cases, or whether that is up to the user to determine.

**Comment by jakewilliami** [comment]
Closing as this seems to be accounted for and I didn't realise.  Apologies for that!
